rapthor/lib/screen.py

```diff
@@ -536,8 +545,19 @@ def fit(self):
             soltab_amp = solset.getSoltab(self.input_amplitude_soltab_name)
 
         # Input data are [time, freq, ant, dir, pol] for slow amplitudes
-        # and [time, freq, ant, dir] for fast phases (scalarphase).
+        # and [time, freq, ant, dir] for fast phases (scalarphase). We reference
+        # the phases to the station with the least amount of flagged solutions,
+        # drawn from the first 10 stations (to ensure it is fairly central)
+        weights = soltab_ph.getValues(retAxesVals=False, weight=True)
+        weights = np.sum(weights, axis=tuple([i for i, axis_name in
+                                              enumerate(soltab_ph.getAxesNames())
+                                              if axis_name != 'ant']), dtype=np.float)
+        ref_ind = np.where(weights[0:10] == np.max(weights[0:10]))[0][0]
```

Comment:
This block is (AFAICS) identical to lines 341-345. You might want to refactor this into a function.

Reply:
Good idea, will do.